vignettes/atlascreation.Rmd

```diff
@@ -0,0 +1,99 @@
+
+## Generating Reference Matrices
+
+Individual reference matrices can be generated which show average gene expression values per individual cell type from a scRNA-seq experiment. In order to generate the reference matrices, the processed data from GEO such as the counts matrix and metadata table will be loaded in. The counts matrix will be checked for containing raw counts (counts matrix may have been previously log or scalar normalized). This is done with the help of a `check_raw_counts()` function. After the counts matrix is checked to contain raw counts, the data will be normalized with the help of the `NormalizeData()` function from Seurat. From there, the counts matrix and metadata table will be inputted as arguments into the clustifyr function `average_clusters()` which will output the reference matrix. The matrix can then be stored into an RDS file.
+
+The `check_raw_counts()` function takes in a valid gene expression counts matrix and a constant max_log_value of 50. It works by checking all of the values in the gene expression matrix for integers indicating that it is raw counts matrix. However, if doubles are found, a new series of checks begin. If a counts value appears to be greater than the max_log_value, the matrix may be considered scalar normalized and if there is no value greater than the max_log_value and the counts matrix contains doubles, it is considered to be log normalized. 
+
+For this vignette, two average gene expression reference matrices will be generated and concatenated into a small reference matrix. The studies used will be GSE113049 (Single Cell RNA Sequencing Identifies TGFβ as a Key Regenerative Cue following LPS-induced Lung Injury) and GSE124952 (Cell type-specific transcriptional programs in mouse prefrontal cortex during adolescence and addiction). 
+
+```{r Reference Matrix Generation GSE113049}
+#An example of reference matrix generation for GEO record GSE113049
+library(dplyr)
+library(Seurat)
+library(clustifyr)
+library(tidyverse)
+
+mat_Lung <- read_tsv("ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE113nnn/GSE113049/suppl/GSE113049_count_matrix.tsv.gz", skip = 1, col_names = F)
+ids_mat_Lung <- read_tsv("ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE113nnn/GSE113049/suppl/GSE113049_count_matrix.tsv.gz", n_max = 1, col_names = F)
+mat_Lung <- mat_Lung %>%
+  column_to_rownames('X1')
+mat_Lung[1:5, 1:5]
+colnames(mat_Lung) <- ids_mat_Lung[1,] %>% unlist() %>% as.vector()
+
+meta_Lung <- read_tsv("ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE113nnn/GSE113049/suppl/GSE113049_cell_metadata.tsv.gz")
+meta_Lung
+sum(colnames(mat_Lung) %in% meta_Lung$cell_type)
+ncol(mat_Lung)
+
+clustifyr::check_raw_counts(as.matrix(mat_Lung))
```

Comment:
this just prints to console also. Might be worth checking the output from the function, otherwise it is not necessary